js/modules/k6/k6_test.go

```diff
@@ -501,3 +525,43 @@ func TestCheckTags(t *testing.T) {
 		}, sample.Tags.Map())
 	}
 }
+
+type testCase struct {
+	samples     chan metrics.SampleContainer
+	testRuntime *modulestest.Runtime
+}
+
+func testCaseRuntime(t testing.TB) *testCase {
```

Comment:
This seems useful, but don't we already have similar helpers? `checkTestRuntime()` is very similar.

Can we reuse an existing helper, and even better, move it to somewhere where it's usable by other packages? This is a generic helper that I'm sure we're duplicating elsewhere, as this setup boilerplate is very common.

Reply:
I need more stuff than what is available in `checkTestRuntime`  - specifically cancelling the context.

I really didn't want to touch `checkTestRuntime` as this change "ripples" through all the tests using it. And I also didn't want this bugfix PR to be 90% me refactoring test code.

> This is a generic helper that I'm sure we're duplicating elsewhere, as this setup boilerplate is very commo

We are but in every case we need something additional for each module separately. 

As part of the ESM saga I would like to be able to test the module loading and resolution in tc39 and also to be usable from other k6 modules (and extensions as consequence). As part of that I will likely try to refactor all of the helpers through k6 modules at least a little bit, hopefully most will be pretty straightforward :crossed_fingers: 

But that will be a *huge* PR or a bunch of pretty big ones as in most cases this helpers are used a *lot* and then their result(s) are used even more.

Reply:
https://github.com/grafana/k6/pull/2946 for just how many line changes the refactor of just this, *really* small module,  are.

Reply:
Thanks! Yeah, I imagined it wouldn't be a short diff, but still, it DRYs things up considerably.